### ISSUE TITLE
some question about the ‘train.py’

### ISSUE THREAD

**Issue by Herbie007** [milesial/Pytorch-UNet#197]
Hello, I have tried to run your code on the Total Text dataset
However, during the training time, I see in the tensorboard the pred image are always all black.
then I found in the train.py, there is no sigmoid function before calculating loss :
```
masks_pred = net(imgs) 
loss = criterion(masks_pred, true_masks)
```
but there is a  sigmoid function before adding image to tensorboard :
```
if net.n_classes == 1:
    writer.add_images('masks/true', true_masks, global_step)
    writer.add_images('masks/pred', torch.sigmoid(masks_pred) > 0.5, global_step)
```

I wonder if there should be a sigmoid function before calculating loss ?
or should I remove the sigmoid function before adding image to tensorboard ?

**Comment by milesial** [comment]
You should not change anything.

The loss is either `CrossEntropyLoss` or `BCEWithLogitsLoss`, in both cases logits are required and the criterion applies the normalization.
For the tensorboard visualization, "sigmoid then threshold" is used because that's what is used for prediction with binary masks.

**Comment by Herbie007** [comment]
I see. Thank you so much ! @milesial
